rasa/core/channels/slack.py

```diff
@@ -348,19 +385,36 @@ def blueprint(
 
                 elif self._is_user_message(output):
                     metadata = self.get_metadata(request)
-                    return await self.process_message(
-                        request,
-                        on_new_message,
-                        self._sanitize_user_message(
-                            output["event"]["text"], output["authed_users"]
-                        ),
-                        output.get("event").get("user"),
-                        metadata,
-                    )
+                    if (
+                        self._is_direct_message(output)
+                        or self._is_app_mention(output)
+                        or metadata["out_channel"] == self.slack_channel
+                    ):
+                        return await self.process_message(
+                            request,
+                            on_new_message,
+                            text=self._sanitize_user_message(
+                                metadata["text"], metadata["users"]
+                            ),
+                            sender_id=metadata["sender"],
+                            metadata=metadata,
+                        )
+                    else:
+                        return response.text(
+                            "Received message on unsupported channel: {}".format(
+                                metadata["out_channel"]
+                            )
+                        )
 
             return response.text("Bot message delivered")
 
         return slack_webhook
 
-    def get_output_channel(self) -> OutputChannel:
-        return SlackBot(self.slack_token, self.slack_channel)
+    def get_output_channel(self, channel: Optional[Text] = None) -> OutputChannel:
+        if channel is not None:
```

Comment:
a short version would be 
```suggestion
        channel = channel or self.slack_channel
        return SlackBot(self.slack_token, channel)
```